apiserver/httpcontext/model.go

```diff
@@ -38,6 +38,37 @@ type QueryModelHandler struct {
 // ServeHTTP is part of the http.Handler interface.
 func (h *QueryModelHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	modelUUID := req.URL.Query().Get(h.Query)
+	validateModelAndServe(h.Handler, modelUUID, w, req)
+}
+
+// BucketModelHandler is an http.Handler that associates requests that
+// it handles with a model UUID extracted from a specified query parameter that
+// must be the objects storage :bucket which is formatted 'model-{modelUUID}'.
+// The model UUID can then be extracted using the RequestModel function
+// in this package.
+type BucketModelHandler struct {
+	http.Handler
+	Query string
+}
+
+// ServeHTTP is part of the http.Handler interface.
+func (h *BucketModelHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	bucket := req.URL.Query().Get(h.Query)
+	// Check that bucket starts with 'model-'
+	prefix := bucket[0:6]
+	if prefix != "model-" {
```

Comment:
You're not checking the length of the bucket string here and then doing direct access to the slice, this will cause a panic.

Additionally, just use `HasPrefix` it does all the validation and bound checking.

```suggestion
	if strings.HasPrefix(bucket, bucketPrefix) {
```

Reply:
Thanks for this! it looks way nicer now :D